### DISCUSSION TITLE
Adding automation capabilities in QGIS

### DISCUSSION THREAD

**Discussion by MyGIS** [GeoCat/qgis-bridge-plugin#156] [Ideas]
Hi, Would it be possible to integrate GeoCAT bridge with the QGIS Model Designer ?
Thanks.

**Comment by GeoSander** [comment]
Hi @MyGIS,

FYI, Bridge does have a (rudimentary) _Processing_ tool, which can also be used in the QGIS Model Designer.

![image](https://user-images.githubusercontent.com/8972335/231117128-10aec79f-fa01-4837-90df-29c03e6242df.png)

In the latest QGIS versions, third-party providers have been deactivated by default, so you will have to activate the Bridge tools first under _QGIS Options > Processing > Providers > GeoCat Bridge_ and then check the `Activate` box:

![image](https://user-images.githubusercontent.com/8972335/231117792-77b95f18-d3f4-4a00-953c-1eb2056f1029.png)

Note that the tool options (layer to GeoServer / metadata to GeoNetwork) are not as advanced as in the Bridge dialog. For example, you can only publish to file-based datastores in GeoServer (not PostGIS).
This is something that can of course be improved, but we haven't had anyone asking for it, nor willing to contribute to it (either financially or by means of a pull request).

Sander

**Comment by MyGIS** [comment]
Hi Sander,
This works for me. I will test it out and add comments under a new thread if I have any issues. Thanks!
